framework/integration/testresource/resource.go

```diff
@@ -9,35 +9,20 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
-type Config struct {
-	K8sClient kubernetes.Interface
-	Logger    micrologger.Logger
-	Name      string
-}
-
 type Resource struct {
 	k8sClient kubernetes.Interface
 	logger    micrologger.Logger
 
 	name string
 }
 
-func New(config Config) (*Resource, error) {
-	r := &Resource{
-		k8sClient: config.K8sClient,
-		logger:    config.Logger,
-
-		name: config.Name,
-	}
-
-	return r, nil
-}
-
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	createCount++
```

Comment:
Let us please not go to work on global variables. The resource is an object which has a pointer receiver. It can be and implement whatever we want. SO we can have getters and setters in here as well to have a proper interface. Later you run two tests with this and don't know where the counts come from.

Reply:
I fixed this.